File: frontend/src/test/fixtures/obsidian.ts
export const obsidianCountsFixture = {
  exported: 2,
  unchanged: 3,
  conflicts: 1,
  errors: 0,
  skipped: 4,
  userManaged: 1,
  orphaned: 2,
  deleted: 0,
} as const;

export const obsidianJobFixture = {
  id: 'job-obsidian-1',
  paperId: 'paper-fixture-1',
  jobType: 'obsidian_export',
  sourceMode: null,
  status: 'queued',
} as const;

export const obsidianJobResponseFixture = {
  job: obsidianJobFixture,
  deduplicated: false,
} as const;

export const obsidianStatusFixture = {
  enabled: true,
  vaultConfigured: true,
  writable: true,
  rootFolder: 'Research',
  pdfMode: 'copy',
  lastJob: {
    id: 'job-obsidian-previous',
    paperId: 'paper-fixture-1',
    jobType: 'obsidian_export',
    status: 'succeeded',
  },
  aggregate: obsidianCountsFixture,
} as const;

export const obsidianSyncStatusFixture = {
  ...obsidianStatusFixture,
  lastJob: {
    ...obsidianStatusFixture.lastJob,
    id: 'job-obsidian-sync-previous',
    paperId: null,
    jobType: 'obsidian_sync',
  },
} as const;

export const obsidianTestFixture = { ok: true } as const;

File: backend/app/api/routes/obsidian.py
from __future__ import annotations

from fastapi import APIRouter, Request

from backend.app.api.schemas.obsidian import (
    ObsidianExportRequest,
    ObsidianSyncRequest,
)
from backend.app.domain import ObsidianDisabledError


def create_obsidian_router() -> APIRouter:
    router = APIRouter()

    async def effective_settings(request: Request):
        service = request.app.state.container.legacy.settings
        return await service.obsidian()

    async def require_enabled(request: Request):
        settings = await effective_settings(request)
        if not settings.enabled:
            raise ObsidianDisabledError()
        return settings

    @router.get("/obsidian/status")
    async def status(request: Request) -> dict[str, object]:
        await effective_settings(request)
        service = getattr(request.app.state.container, "obsidian_jobs", None)
        if service is None:
            settings = await effective_settings(request)
            return {
                "enabled": settings.enabled,
                "vaultConfigured": bool(settings.vault_path),
                "writable": False,
                "rootFolder": settings.root_folder,
                "pdfMode": settings.pdf_mode,
                "lastJob": None,
                "aggregate": {
                    key: 0
                    for key in (
                        "exported", "unchanged", "conflicts", "errors",
                        "skipped", "userManaged", "orphaned", "deleted",
                    )
                },
            }
        return await service.status()

    @router.post("/papers/{paper_id}/exports/obsidian", status_code=202)
    async def export_paper(
        paper_id: str, body: ObsidianExportRequest, request: Request
    ) -> dict[str, object]:
        await require_enabled(request)
        result = await request.app.state.container.obsidian_jobs.enqueue_export(
            paper_id,
            dry_run=body.dryRun,
        )
        return {"job": _job_summary(result.job), "deduplicated": result.deduplicated}

    @router.post("/obsidian/sync", status_code=202)
    async def sync(
        body: ObsidianSyncRequest, request: Request
    ) -> dict[str, object]:
        await require_enabled(request)
        result = await request.app.state.container.obsidian_jobs.enqueue_sync(
            dry_run=body.dryRun,
            apply_cleanup=body.applyCleanup,
            cleanup_plan_sha=body.cleanupPlanSha,
        )
        return {"job": _job_summary(result.job), "deduplicated": result.deduplicated}

    @router.post("/obsidian/test")
    async def test(request: Request) -> dict[str, object]:
        await require_enabled(request)
        return {"ok": await request.app.state.container.obsidian_jobs.test_access()}

    return router


def _job_summary(job: object) -> dict[str, object]:
    job_type = getattr(job, "job_type")
    status = getattr(job, "status")
    source_mode = getattr(job, "source_mode")
    return {
        "id": getattr(job, "id"),
        "paperId": getattr(job, "paper_id"),
        "jobType": getattr(job_type, "value", job_type),
        "sourceMode": getattr(source_mode, "value", source_mode),
        "status": getattr(status, "value", status),
    }


__all__ = ["create_obsidian_router"]
